sdk/confidentialledger/azure-confidentialledger/azure/confidentialledger/receipt/_claims_models.py

```diff
@@ -0,0 +1,156 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+"""Models for application claims."""
+
+from typing import Any, Dict, Optional, Union
+
+
+class LedgerEntryClaim:
+    """
+    LedgerEntryClaim represents an Application Claim derived from ledger entry data.
+    """
+
+    def __init__(
+        self,
+        protocol: str,
+        collectionId: str,
+        contents: str,
+        secretKey: str,
+    ):
+        """
+        :keyword protocol: The protocol used to compute the claim.
+        :paramtype protocol: str
+
+        :keyword collectionId: The collection ID of the ledger entry.
+        :paramtype collectionId: str
+
+        :keyword contents: The contents of the ledger entry.
+        :paramtype contents: str
+
+        :keyword secretKey: The secret key used to compute the claim digest.
+        :paramtype secretKey: str
+        """
+        self.protocol = protocol
+        self.collectionId = collectionId
+        self.contents = contents
+        self.secretKey = secretKey
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, LedgerEntryClaim):
+            return NotImplemented
+
+        return (
+            self.protocol == other.protocol
+            and self.collectionId == other.collectionId
+            and self.contents == other.contents
+            and self.secretKey == other.secretKey
+        )
+
+    @classmethod
+    def from_dict(cls, ledger_entry_claim_dict: Dict[str, Any]):
+        """Create a new instance of this class from a dictionary."""
+
+        return cls(
+            protocol=ledger_entry_claim_dict["protocol"],
+            collectionId=ledger_entry_claim_dict["collectionId"],
+            contents=ledger_entry_claim_dict["contents"],
+            secretKey=ledger_entry_claim_dict["secretKey"],
+        )
+
+
+class ClaimDigest:
+    """
+    ClaimDigest represents an Application Claim in digested form.
+    """
+
+    def __init__(
+        self,
+        protocol: str,
+        value: str,
+    ):
+        """
+        :keyword protocol: The protocol used to compute the claim.
+        :paramtype protocol: str
+
+        :keyword value: The digest of the claim.
+        :paramtype value: str
+        """
+        self.protocol = protocol
+        self.value = value
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, ClaimDigest):
+            return NotImplemented
+
+        return self.protocol == other.protocol and self.value == other.value
+
+    @classmethod
+    def from_dict(cls, ledger_entry_claim_dict: Dict[str, Any]):
+        """Create a new instance of this class from a dictionary."""
+
+        return cls(
+            protocol=ledger_entry_claim_dict["protocol"],
+            value=ledger_entry_claim_dict["value"],
+        )
+
+
+class ApplicationClaim:
+    """
+    ApplicationClaim represents a claim of a ledger application.
+    """
+
+    def __init__(
```

Comment:
Even though this `__init__` is more complicated, you can still use `@dataclass`, which will generate an `__eq__` for you and skip `__init__` if you defined one already

Reply:
Done